python_modules/dagster/dagster/_daemon/run_coordinator/queued_run_coordinator_daemon.py

```diff
@@ -231,17 +231,22 @@ def _get_runs_to_dequeue(
 
         self._logger.info(
             f"Retrieved %d queued runs, checking limits.{locations_clause}",
-            len(queued_runs),
+            len(sorted_run_ids),
         )
 
         # place in order
-        sorted_runs = self._priority_sort(queued_runs)
         tag_concurrency_limits_counter = TagConcurrencyLimitsCounter(
             tag_concurrency_limits, in_progress_runs
         )
 
         batch: List[DagsterRun] = []
-        for run in sorted_runs:
+        for run_id in sorted_run_ids:
+            runs = instance.get_runs(RunsFilter(run_ids=[run_id]))
```

Comment:
i think its worth the impl time / complexity to do some basic version of batched fetching. I can imagine round trip time between each individual run being rough for some oss deployments. 

* some fixed batch size 10/25/50
* `max_runs_to_launch` for the first batch, then fixed 10/25 after that